src/PlayerbotAI.cpp

```diff
@@ -434,6 +441,126 @@ void PlayerbotAI::UpdateAIGroupMaster()
     }
 }
 
+void PlayerbotAI::UpdatePvPGearSwap(uint32 elapsed)
+{
+    if (!bot || !bot->GetSession() || !bot->IsInWorld())
+        return;
+
+    bool isPvpContext = bot->InBattleground() || bot->InArena();
+
+    // Detect transitions world <-> BG/Arena
+    if (isPvpContext != lastPvpContext)
+    {
+        lastPvpContext = isPvpContext;
+        pvpGearSwapPending = true;
+        pvpGearSwapCooldown = 0;  // try immediately
+    }
+
+    if (!pvpGearSwapPending)
+        return;
+
+    if (pvpGearSwapCooldown)
+    {
+        if (pvpGearSwapCooldown > elapsed)
+        {
+            pvpGearSwapCooldown -= elapsed;
+            return;
+        }
+        pvpGearSwapCooldown = 0;
+    }
+
+    // Can't change equipment in combat / teleport; retry later.
+    if (!bot->IsAlive() || bot->IsInCombat() || bot->IsBeingTeleported() || bot->GetSession()->isLogingOut() ||
+        bot->IsDuringRemoveFromWorld())
+    {
+        pvpGearSwapCooldown = 5000;
+        return;
+    }
+
+    // Important: "alt" bots may be temporarily controlled by someone else.
+    // We must not decide this based on master's account.
+    // The module already uses RandomAccountList to distinguish real alts from bot accounts
+    // (see AutoGearAction / PlayerbotAIConfig::IsInRandomAccountList).
+    // - accounts in RandomAccountList => bot accounts (allowed to generate gear)
+    // - all other accounts            => real player alts (bag-only)
+    bool isBotAccount = sRandomPlayerbotMgr->IsRandomBot(bot) ||
+                        sPlayerbotAIConfig->IsInRandomAccountList(bot->GetSession()->GetAccountId());
+    bool bagOnly = !isBotAccount;
```

Comment:
Why does this exist? bagOnly is only called right under this so that if might as well
```cpp
if (!isBotAccount)
```
rather than this definition followed by
```cpp
if (bagOnly)
```